mesonbuild/environment.py

```diff
@@ -395,6 +395,27 @@ def detect_cpu(compilers: CompilersDict) -> str:
     # detect_cpu_family() above.
     return trial
 
+kernel_mappings = {'freebsd': 'freebsd',
+                   'openbsd': 'openbsd',
+                   'netbsd': 'netbsd',
+                   'windows': 'nt',
+                   'android': 'linux',
+                   'linux': 'linux',
+                   'cygwin': 'nt',
+                   'darwin': 'xnu',
+                   'sunos': 'sunos',
+                   'dragonfly': 'dragonfly',
```

Comment:
Aside: I know some people think that the value "dragonfly" for system() was a mistake as it prevents checking for .endswith('bsd')

Maybe this is something we can take the opportunity to fix?

Reply:
If we could change `system`, cure. With just the kernel I'm not so sure. It would also be confusing that `dragonfly` is the only bsd whose kernel name is different from the system name. Can we ask df people which one they'd prefer?

Reply:
a `DeprecatedValue()` object? I kinda like that, it would compare against both any deprecated values and the current value, but with a FeatureDeprecated if only the old values matches... hmmm

Reply:
I wonder how that functionality would be documented... :thinking:

> It would also be confusing that `dragonfly` is the only bsd whose kernel name is different from the system name. Can we ask df people which one they'd prefer?

I am not sure how to contact all the people who I've seen discuss it. But @Gio is one of those people -- and even submitted #10359 to change that, which we didn't want to do because, to quote you on that very PR,

> We have also committed to the fact that system names are immutable. This is why `darwin` is still a value rather than `macos`, `ios` et al.

Actually, also based on that PR, we got #10360 which is closely related to this PR. It asked for a `.system_family()` method.

Reply:
@dwfreed might have an opinion too?